### ISSUE TITLE
The inbound connection has been dropped

### ISSUE THREAD

**Issue by kim-sardine** [kubeflow/mpi-operator#299]
Hi, I'm running kubeflow (v.1.2.0) on AWS EKS

When I ran [tensorflow-benchmarks.yaml](https://github.com/kubeflow/mpi-operator/blob/master/examples/v1alpha2/tensorflow-benchmarks.yaml), I got below error.
(using 2 workers with 1 gpu for each)

```
+ POD_NAME=tensorflow-benchmarks-2-worker-0
+ [ t = - ]
+ shift
+ /opt/kube/kubectl cp /opt/kube/hosts tensorflow-benchmarks-2-worker-0:/etc/hosts_of_nodes
+ POD_NAME=tensorflow-benchmarks-2-worker-1
+ [ t = - ]
+ shift
+ /opt/kube/kubectl cp /opt/kube/hosts tensorflow-benchmarks-2-worker-1:/etc/hosts_of_nodes
Defaulting container name to tensorflow-benchmarks.
Defaulting container name to tensorflow-benchmarks.
+ /opt/kube/kubectl exec tensorflow-benchmarks-2-worker-0 -- /bin/sh -c cat /etc/hosts_of_nodes >> /etc/hosts &&     ### BLAH_BLAH
+ /opt/kube/kubectl exec tensorflow-benchmarks-2-worker-1 -- /bin/sh -c cat /etc/hosts_of_nodes >> /etc/hosts &&     ### BLAH_BLAH

### Lots of warnings....

--------------------------------------------------------------------------
Open MPI detected an inbound MPI TCP connection request from a peer
that appears to be part of this MPI job (i.e., it identified itself as
part of this Open MPI job), but it is from an IP address that is
unexpected.  This is highly unusual.

The inbound connection has been dropped, and the peer should simply
try again with a different IP interface (i.e., the job should
hopefully be able to continue).

  Local host:          tensorflow-benchmarks-2-worker-0
  Local PID:           26
  Peer hostname:       tensorflow-benchmarks-2-worker-1 ([[33442,1],2])
  Source IP of socket: 127.0.0.6
  Known IPs of peer:   
	192.168.12.64
--------------------------------------------------------------------------
```

It looks like `tensorflow-benchmarks-2-worker-0` is trying to use `127.0.0.6`, not `192.168.12.64`.

when I run `ifconfig` on `tensorflow-benchmarks-2-worker-0`, I get

```
root@tensorflow-benchmarks-2-worker-0:/tensorflow/benchmarks# ifconfig
eth0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 9001
        inet 192.168.12.17  netmask 255.255.255.255  broadcast 0.0.0.0
        ether aa:b5:2a:70:50:b5  txqueuelen 0  (Ethernet)
        RX packets 160462  bytes 84160213 (84.1 MB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 158802  bytes 130447055 (130.4 MB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

lo: flags=73<UP,LOOPBACK,RUNNING>  mtu 65536
        inet 127.0.0.1  netmask 255.0.0.0
        loop  txqueuelen 1000  (Local Loopback)
        RX packets 93667  bytes 257861331 (257.8 MB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 93667  bytes 257861331 (257.8 MB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
```

**Comment by kim-sardine** [comment]
My MPIJob spec is

```yaml
apiVersion: kubeflow.org/v1alpha2
kind: MPIJob
metadata:
  name: tensorflow-benchmarks-2
spec:
  slotsPerWorker: 1
  cleanPodPolicy: Running
  mpiReplicaSpecs:
    Launcher:
      replicas: 1
      template:
         spec:
           containers:
           - image: mpioperator/tensorflow-benchmarks:latest
             name: tensorflow-benchmarks
             command:
             - mpirun
             - --allow-run-as-root
             - -np
             - "2"
             - -bind-to
             - none
             - -map-by
             - slot
             - -x
             - NCCL_DEBUG=INFO
             - -x
             - LD_LIBRARY_PATH
             - -x
             - PATH
             - -mca
             - pml
             - ob1
             - -mca
             - btl
             - ^openib
             - python
             - scripts/tf_cnn_benchmarks/tf_cnn_benchmarks.py
             - --model=resnet101
             - --batch_size=64
             - --variable_update=horovod
    Worker:
      replicas: 2
      template:
        spec:
          containers:
          - image: mpioperator/tensorflow-benchmarks:latest
            name: tensorflow-benchmarks
            resources:
              limits:
                nvidia.com/gpu: 1
```

**Comment by kim-sardine** [comment]
SOLVED. It was namespace problem.

tl;dr - do not use kubeflow generated namespace

I was creating above MPIJob in kubeflow pipeline-generated namespace(`admin`). and It turns out that only single-node MPIJob (`replicas: 1`) is available in here.

I confirmed that multi-node MPIJob works well in newly-created custom namespaces or default namespace.

I'm not sure exactly why multi-node MPIJob on kubeflow-generated namespace doen't work.
maybe It's because of its labels (`app.kubernetes.io/part-of=kubeflow-profile`, `istio-injection=enabled`, `katib-metricscollector-injection=enabled`, `serving.kubeflow.org/inferenceservice=enabled`)
